src/main/java/net/gensokyoreimagined/motoori/KosuzuUnderstandsEverything.java

```diff
@@ -112,23 +113,28 @@ private void onServerChatSend(PacketEvent event) {
 
         var json = message.getJson();
         var component = JSONComponentSerializer.json().deserialize(json); // Adventure API from raw JSON
-        var text = parser.getTextMessage(component, player);
+        var text = PlainTextComponentSerializer.plainText().serialize(component);
+        var actualMessage = parser.getTextMessage(text, player);
 
-        if (text != null) {
-            var uuid = database.addMessage(json, text);
+        TextComponent newComponent;
+        if (actualMessage != null) {
+            actualMessage = parser.removeUnwantedSyntax(message);
+            var uuid = database.addMessage(json, actualMessage);
 
-            var newComponent = component.hoverEvent(
+            newComponent = component.hoverEvent(
                     Component
                         .text(database.getTranslation("translate.hover", database.getUserDefaultLanguage(player.getUniqueId())))
                         .color(NamedTextColor.GRAY)
                 )
                 .clickEvent(
                     ClickEvent.runCommand("/kosuzu translate " + uuid.toString())
                 );
-
-            var newJson = JSONComponentSerializer.json().serialize(newComponent);
-            packet.getChatComponents().write(0, WrappedChatComponent.fromJson(newJson));
+        } else {
+            newComponent = component.content(parser.removeUnwantedSyntax(text));
```

Comment:
Not sure why we need to invoke `parser.removeUnwantedSyntax` in both branches? We need to strip those symbols out anyways regardless if the message is translatable or not.

Reply:
It has to do with the order of what Kosuzu wants to do. It determines if a message should be stored in the translation cache with the same function that it uses to get the message content. As a result, there then become two different variables: one for the message content, and one for the translatable message (which could be `null`). Obviously, we don't want to try to remove substrings from `null`, which is why the branch is needed.

Reply:
Sounds good.